provider/win_dump_utils.py

```diff
@@ -50,19 +50,23 @@ def generate_mem_dump(test, params, vm):
 
     output = vm.monitor.human_monitor_cmd('dump-guest-memory -w %s'
                                           % dump_file)
-    if output and "warning" not in output:
-        test.fail("Save dump file failed as: %s" % output)
-    else:
-        cmd = "ls -l %s | awk '{print $5}'" % dump_file
-        dump_size = int(process.getoutput(cmd))
-        if dump_size == 0:
-            test.fail("The size of dump file is %d" % dump_size)
-
-    dump_name_zip = "%s.zip" % dump_name
-    process.system("cd %s && zip %s %s" % (tmp_dir, dump_name_zip,
-                                           dump_name), shell=True)
-    dump_file_zip = tmp_dir + "/" + dump_name_zip
-    return dump_file, dump_file_zip
+    try:
+        if output and "warning" not in output:
+            test.fail("Save dump file failed as: %s" % output)
+        else:
+            cmd = "ls -l %s | awk '{print $5}'" % dump_file
+            dump_size = int(process.getoutput(cmd))
+            if dump_size == 0:
+                test.fail("The size of dump file is %d" % dump_size)
+        dump_name_zip = "%s.zip" % dump_name
+        process.system("cd %s && zip %s %s" % (tmp_dir, dump_name_zip,
+                                               dump_name), shell=True)
+        dump_file_zip = tmp_dir + "/" + dump_name_zip
+        return dump_file, dump_file_zip
+    except:
+        pass
+    finally:
+        process.system("rm -rf %s" % dump_file)
```

Comment:
It is not safe that we clean the dmp file during the `generate_mem_dump` function, the dmp file probably need to transfer to VM in the next step.So we can not add clean step to this function.

And for the fwcfg testing, we already clean the dmp file before the end of test.
FYI:https://github.com/autotest/tp-qemu/blob/66ab2d732ce09a2f1b221adc6bb3bff657e67d95/qemu/tests/fwcfg.py#L75